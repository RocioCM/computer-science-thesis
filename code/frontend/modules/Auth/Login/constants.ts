import { FormBuilderField, INPUT_TYPES } from '@/common/hooks/useForm/types';
import * as validators from '@/common/hooks/useForm/validators';

export const LOGIN_FORM_STRUCT = [
  {
    name: 'email',
    required: true,
    default: '',
    validators: validators.isEmail,
  },
  {
    name: 'password',
    required: true,
    default: '',
  },
];

export const LOGIN_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'email',
    placeholder: '',
    label: 'Correo electrónico',
  },
  {
    type: INPUT_TYPES.password,
    name: 'password',
    placeholder: '',
    label: 'Contraseña',
  },
];
