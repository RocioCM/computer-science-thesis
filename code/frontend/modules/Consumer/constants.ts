import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

export const SEARCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'search',
    required: false,
    default: '',
  },
];

export const SEARCH_FORM_INPUTS: FormBuilderField[] = [
  {
    name: 'search',
    type: INPUT_TYPES.search,
    label: 'Código de seguimiento',
    placeholder: 'ABC1234XYZ',
  },
];

// ------- Waste bottle recycle form ------- //

export const RECYCLE_FORM_STRUCT: FieldData[] = [
  {
    name: 'trackingCode',
    required: true,
    default: '',
  },
  {
    name: 'recyclerUid',
    required: true,
    default: '',
  },
];

export const RECYCLE_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'trackingCode',
    placeholder: 'Ingresá el código de seguimiento',
    label: 'Código de seguimiento',
  },
  {
    type: INPUT_TYPES.autocomplete,
    name: 'recyclerUid',
    placeholder: 'Ingresá el nombre o correo del reciclador',
    label: 'Reciclador/a',
  },
];
