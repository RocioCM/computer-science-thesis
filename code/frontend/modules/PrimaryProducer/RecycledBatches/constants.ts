import { arrayOf, validators } from '@/common/hooks/useForm';
import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

export const SEARCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'search',
    required: false,
    default: '',
  },
];

export const SEARCH_FORM_INPUTS: FormBuilderField[] = [
  {
    name: 'search',
    type: INPUT_TYPES.text,
    label: 'Código de seguimiento',
    placeholder: 'ABC1234XYZ',
  },
];

// ----------- Recycling Batch Form ------------- //

export const BATCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'weight',
    required: true,
    default: '',
  },
  {
    name: 'size',
    required: true,
    default: '',
  },
  {
    name: 'materialType',
    required: true,
    default: '',
  },
  {
    name: 'composition',
    required: true,
    default: [{ name: '', amount: '', measureUnit: '' }],
    structure: arrayOf([
      {
        name: 'name',
        required: true,
        default: '',
      },
      {
        name: 'amount',
        required: true,
        default: '',
        validators: validators.isPositiveNumber,
      },
      {
        name: 'measureUnit',
        required: true,
        default: '',
      },
    ]),
  },
  {
    name: 'extraInfo',
    required: false,
    default: '',
  },
  {
    name: 'wasteBottleIds',
    required: false,
    default: [],
  },
  {
    name: 'createdAt',
    required: true,
    disabled: (form) => !!form.id,
    default: new Date().toISOString().split('T')[0],
  },
];

export const BATCH_FORM_INPUTS_1: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'weight',
    placeholder: 'Ingresá el peso del lote de material',
    label: 'Peso del fardo (en kg)',
  },
  {
    type: INPUT_TYPES.text,
    name: 'size',
    placeholder: 'Ingresá el tamaño del fardo',
    label: 'Tamaño del fardo (con unidades)',
  },
  {
    type: INPUT_TYPES.text,
    name: 'materialType',
    placeholder: 'Ingresá el tipo de material del lote',
    label: 'Tipo de material',
  },
  {
    type: INPUT_TYPES.date,
    name: 'createdAt',
    placeholder: 'Seleccioná la fecha de producción',
    label: 'Fecha de producción',
    max: new Date().toISOString().split('T')[0],
  },
  {
    type: INPUT_TYPES.textarea,
    name: 'extraInfo',
    placeholder: 'Ingresá observaciones adicionales',
    label: 'Observaciones adicionales (opcional)',
    containerClassName: 'col-span-2',
  },
];

export const BATCH_COMPOSITION_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'name',
    placeholder: 'Ingresá el nombre del material',
    label: 'Nombre del material',
  },
  {
    type: INPUT_TYPES.number,
    name: 'amount',
    placeholder: 'Ingresá la cantidad de material',
    label: 'Cantidad de material',
  },
  {
    type: INPUT_TYPES.text,
    name: 'measureUnit',
    placeholder: 'Ingresá la unidad de medida de la cantidad de material',
    label: 'Unidad de medida',
  },
];

// ------- Sell form ------- //

export const SELL_FORM_STRUCT: FieldData[] = [
  {
    name: 'buyerUid',
    required: true,
    default: '',
  },
];

export const SELL_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.autocomplete,
    name: 'buyerUid',
    placeholder: 'Ingresá el nombre o correo del comprador',
    label: 'Comprador',
  },
];

//// TODO: remove unused
