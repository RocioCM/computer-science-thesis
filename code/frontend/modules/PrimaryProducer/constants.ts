import { arrayOf, object, validators } from '@/common/hooks/useForm';
import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

// type Material = {
//   name: string;
//   amount: number;
//   measureUnit: string;
// };

// type Bottle = {
//   weight: number; // uint256
//   color: string;
//   thickness: number; // uint256
//   composition: Material[];
//   shapeType: string;
//   originLocation: string;
//   extraInfo: string;
// };

// export type BottleBatch = {
//   id: number; // uint256
//   quantity: number; // uint256
//   bottleType: Bottle;
//   soldQuantity: number; // uint256
//   owner: string; // Ethereum address
//   createdAt: string; // Timestamp as a string
// };

export const BATCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'quantity',
    required: true,
    default: '',
    validators: validators.isNaturalNumber,
  },
  {
    name: 'bottleType',
    required: true,
    structure: object([
      {
        name: 'weight',
        required: true,
        default: '',
        validators: validators.isPositiveNumber,
      },
      {
        name: 'color',
        required: true,
        default: '',
      },
      {
        name: 'composition',
        required: true,
        default: [{ name: '', amount: '', measureUnit: '' }],
        structure: arrayOf([
          {
            name: 'name',
            required: true,
            default: '',
          },
          {
            name: 'amount',
            required: true,
            default: '',
            validators: validators.isPositiveNumber,
          },
          {
            name: 'measureUnit',
            required: true,
            default: '',
          },
        ]),
      },
      {
        name: 'thickness',
        required: true,
        default: '',
        validators: validators.isPositiveNumber,
      },
      {
        name: 'shapeType',
        required: true,
        default: '',
      },

      {
        name: 'originLocation',
        required: true,
        default: '',
      },
      {
        name: 'extraInfo',
        required: false,
        default: '',
      },
    ]),
  },
  {
    name: 'createdAt',
    required: true,
    disabled: (form) => !!form.id,
    default: new Date().toISOString().split('T')[0],
  },
];

export const BATCH_FORM_INPUTS_1: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'quantity',
    placeholder: 'Ingresá la cantidad de envases',
    label: 'Cantidad de envases',
  },
  {
    type: INPUT_TYPES.number,
    name: 'bottleType.weight',
    placeholder: 'Ingresá el peso por envase (gramos)',
    label: 'Peso por envase (en gramos)',
  },
  {
    type: INPUT_TYPES.dropdown,
    name: 'bottleType.color',
    placeholder: 'Seleccioná el color del envase',
    label: 'Color del envase',
    dropdownOptions: [
      { value: 'transparente', label: 'Transparente' },
      { value: 'verde', label: 'Verde' },
      { value: 'ambar', label: 'Ámbar' },
      { value: 'otro', label: 'Otro' },
    ],
  },
  {
    type: INPUT_TYPES.number,
    name: 'bottleType.thickness',
    placeholder: 'Ingresá el espesor (mm)',
    label: 'Espesor (en milímetros)',
  },
  {
    type: INPUT_TYPES.text,
    name: 'bottleType.shapeType',
    placeholder: 'Ingresá el tipo/forma del envase',
    label: 'Tipo/forma del envase',
  },
];

export const BATCH_COMPOSITION_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'name',
    placeholder: 'Ingresá el nombre del material',
    label: 'Nombre del material',
  },
  {
    type: INPUT_TYPES.number,
    name: 'amount',
    placeholder: 'Ingresá la cantidad de material',
    label: 'Cantidad de material',
  },
  {
    type: INPUT_TYPES.text,
    name: 'measureUnit',
    placeholder: 'Ingresá la unidad de medida de la cantidad de material',
    label: 'Unidad de medida',
  },
];

export const BATCH_FORM_INPUTS_2: FormBuilderField[] = [
  {
    type: INPUT_TYPES.date,
    name: 'createdAt',
    placeholder: 'Seleccioná la fecha de producción',
    label: 'Fecha de producción',
    max: new Date().toISOString().split('T')[0],
  },
  {
    type: INPUT_TYPES.text,
    name: 'bottleType.originLocation',
    placeholder: 'Ingresá la localización de producción',
    label: 'Localización de producción',
  },
  {
    type: INPUT_TYPES.textarea,
    name: 'bottleType.extraInfo',
    placeholder: 'Ingresá observaciones adicionales',
    label: 'Observaciones adicionales (opcional)',
    containerClassName: 'col-span-2',
  },
];

// ------- Sell form ------- //

export const SELL_FORM_STRUCT: FieldData[] = [
  {
    name: 'quantity',
    required: true,
    default: '',
    validators: validators.isNaturalNumber,
  },
  {
    name: 'buyerUid',
    required: true,
    default: '',
  },
];

export const SELL_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'quantity',
    placeholder: 'Ingresá la cantidad de envases a vender',
    label: 'Cantidad de envases a vender',
  },
  {
    type: INPUT_TYPES.autocomplete,
    name: 'buyerUid',
    placeholder: 'Ingresá el nombre o correo del comprador',
    label: 'Comprador',
  },
];

// ------- Recycle form ------- //

export const RECYCLE_FORM_STRUCT: FieldData[] = [
  {
    name: 'quantity',
    required: true,
    default: '',
    validators: validators.isNaturalNumber,
  },
];

export const RECYCLE_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'quantity',
    placeholder: 'Ingresá la cantidad de envases a reciclar',
    label: 'Cantidad de envases a reciclar',
  },
];
