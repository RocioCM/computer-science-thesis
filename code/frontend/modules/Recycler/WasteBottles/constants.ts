import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

export const TABS = {
  available: 'Disponibles',
  recycled: 'Recicladas',
};

// ------- Search form ------- //

export const SEARCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'search',
    required: false,
    default: '',
  },
];

export const SEARCH_FORM_INPUTS: FormBuilderField[] = [
  {
    name: 'search',
    type: INPUT_TYPES.text,
    label: 'CÃ³digo de seguimiento',
    placeholder: 'ABC1234XYZ',
  },
];

// ------- Recycle form ------- //

export const ASSIGN_FORM_STRUCT: FieldData[] = [
  {
    name: 'batchId',
    required: true,
    default: null,
  },
];

export const ASSIGN_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.dropdown,
    name: 'batchId',
    placeholder: 'Selecciona un lote',
    label: 'ID del lote',
  },
];
