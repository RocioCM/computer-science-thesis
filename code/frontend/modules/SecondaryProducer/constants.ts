import { validators } from '@/common/hooks/useForm';
import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

export const BATCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'trackingCode',
    required: true,
    default: '',
  },
];

export const BATCH_CODE_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'trackingCode',
    placeholder: 'ABC1234XYZ',
    label: 'CÃ³digo de seguimiento',
  },
];

// ------- Sell form ------- //

export const SELL_FORM_STRUCT: FieldData[] = [
  {
    name: 'quantity',
    required: true,
    default: '',
    validators: validators.isNaturalNumber,
  },
  {
    name: 'buyerUid',
    required: true,
    default: '',
  },
];

export const SELL_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'quantity',
    placeholder: '100',
    label: 'Cantidad de botellas a vender',
  },
  {
    type: INPUT_TYPES.autocomplete,
    name: 'buyerUid',
    placeholder: 'Busca por nombre o correo del comprador',
    label: 'Comprador',
  },
];

// ------- Recycle form ------- //

export const RECYCLE_FORM_STRUCT: FieldData[] = [
  {
    name: 'quantity',
    required: true,
    default: '',
    validators: validators.isNaturalNumber,
  },
];

export const RECYCLE_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'quantity',
    placeholder: '100',
    label: 'Cantidad de envases a reciclar',
  },
];
