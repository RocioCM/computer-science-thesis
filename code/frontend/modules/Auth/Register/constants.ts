import { ROLES_OPTIONS } from '@/common/constants/auth';
import { FormBuilderField, INPUT_TYPES } from '@/common/hooks/useForm/types';
import * as validators from '@/common/hooks/useForm/validators';

export const REGISTER_FORM_STRUCT = [
  {
    name: 'email',
    required: true,
    default: '',
    validators: validators.isEmail,
  },
  {
    name: 'password',
    required: true,
    default: '',
    validators: validators.isPassword,
  },
  {
    name: 'roleId',
    required: true,
    default: null,
  },
];

export const REGISTER_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'email',
    placeholder: 'Ingresá tu email',
    label: 'Correo electrónico',
  },
  {
    type: INPUT_TYPES.password,
    name: 'password',
    placeholder: 'Ingresá tu contraseña',
    label: 'Contraseña',
  },
  {
    type: INPUT_TYPES.dropdown,
    name: 'roleId',
    placeholder: 'Seleccioná tu rol',
    label: 'Rol',
    dropdownOptions: ROLES_OPTIONS,
  },
];
