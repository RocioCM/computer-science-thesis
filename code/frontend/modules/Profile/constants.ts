import { ROLES_OPTIONS } from '@/common/constants/auth';
import { validators } from '@/common/hooks/useForm';
import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

export const PROFILE_FORM_STRUCT: FieldData[] = [
  {
    name: 'email',
    disabled: true,
    default: '',
  },

  {
    name: 'role',
    disabled: true,
    default: null,
  },
  {
    name: 'blockchainId',
    disabled: true,
    default: '',
  },
  {
    name: 'userName',
    default: '',
  },
  {
    name: 'managerName',
    default: '',
  },
  {
    name: 'phone',
    default: '',
    validators: [validators.isPhone],
  },
];

export const PROFILE_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'email',
    label: 'Correo electrónico',
  },
  {
    type: INPUT_TYPES.dropdown,
    name: 'role',
    label: 'Rol',
    dropdownOptions: ROLES_OPTIONS,
  },
  {
    type: INPUT_TYPES.text,
    name: 'blockchainId',
    label: 'Dirección Blockchain',
  },
  {
    type: INPUT_TYPES.text,
    name: 'userName',
    label: 'Nombre de usuario',
  },
  {
    type: INPUT_TYPES.text,
    name: 'managerName',
    label: 'Nombre del Responsable',
  },
  {
    type: INPUT_TYPES.text,
    name: 'phone',
    label: 'Teléfono',
  },
];
