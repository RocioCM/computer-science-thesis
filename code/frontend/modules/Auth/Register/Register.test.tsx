import TestAppWrapper from '@/common/utils/tests';
import RegisterPage from '@/pages/auth/register';
import { render, screen, fireEvent, act } from '@testing-library/react';
import { toast } from 'react-toastify';
import auth from '@/common/libraries/auth';
import { ROLES } from '@/common/constants/auth';

describe('Register Page', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Matches snapshot', () => {
    const { container } = render(
      <TestAppWrapper>
        <RegisterPage />
      </TestAppWrapper>
    );
    expect(container).toMatchSnapshot();
  });

  it('Displays error when registration fails due to conflict', async () => {
    const mockRegisterUser = jest
      .fn()
      .mockResolvedValue({ ok: false, status: 409 });
    (auth as jest.Mock).mockReturnValue({
      registerUser: mockRegisterUser,
    });

    render(
      <TestAppWrapper>
        <RegisterPage />
      </TestAppWrapper>
    );

    const [emailInput, passwordInput] = screen.queryAllByTestId('input');
    const rolesDropdown = screen.getByTestId('dropdown');
    const submitButton = screen.getByRole('button', { name: /crear usuario/i });

    await act(async () => rolesDropdown.click());

    const roleOption = screen.getByText('Reciclador');

    await act(async () => {
      fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
      fireEvent.change(passwordInput, { target: { value: 'password123' } });
      fireEvent.click(roleOption);
    });

    await act(async () => fireEvent.click(submitButton));

    expect(mockRegisterUser).toHaveBeenCalledWith({
      email: 'test@example.com',
      password: 'password123',
      roleId: ROLES.recycler,
    });
    expect(toast.error).toHaveBeenCalledWith(
      'El email ingresado ya está registrado, intentá con otro'
    );
  });

  it('Displays error when registration fails due to unknown error', async () => {
    const mockRegisterUser = jest
      .fn()
      .mockResolvedValue({ ok: false, status: 500 });
    (auth as jest.Mock).mockReturnValue({
      registerUser: mockRegisterUser,
    });

    render(
      <TestAppWrapper>
        <RegisterPage />
      </TestAppWrapper>
    );

    const [emailInput, passwordInput] = screen.queryAllByTestId('input');
    const rolesDropdown = screen.getByTestId('dropdown');
    const submitButton = screen.getByRole('button', { name: /crear usuario/i });

    await act(async () => rolesDropdown.click());

    const roleOption = screen.getByText('Reciclador');

    await act(async () => {
      fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
      fireEvent.change(passwordInput, { target: { value: 'password123' } });
      fireEvent.click(roleOption);
    });

    await act(async () => fireEvent.click(submitButton));

    expect(mockRegisterUser).toHaveBeenCalledWith({
      email: 'test@example.com',
      password: 'password123',
      roleId: ROLES.recycler,
    });
    expect(toast.error).toHaveBeenCalledWith(
      'Ocurrió un error al registrar. Intentá nuevamente'
    );
  });

  it('Displays success message on successful registration', async () => {
    const mockRegisterUser = jest.fn().mockResolvedValue({ ok: true });
    (auth as jest.Mock).mockReturnValue({
      registerUser: mockRegisterUser,
    });

    render(
      <TestAppWrapper>
        <RegisterPage />
      </TestAppWrapper>
    );

    const [emailInput, passwordInput] = screen.queryAllByTestId('input');
    const rolesDropdown = screen.getByTestId('dropdown');
    const submitButton = screen.getByRole('button', { name: /crear usuario/i });

    await act(async () => rolesDropdown.click());

    const roleOption = screen.getByText('Reciclador');

    await act(async () => {
      fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
      fireEvent.change(passwordInput, { target: { value: 'password123' } });
      fireEvent.click(roleOption);
    });

    await act(async () => fireEvent.click(submitButton));

    expect(mockRegisterUser).toHaveBeenCalledWith({
      email: 'test@example.com',
      password: 'password123',
      roleId: ROLES.recycler,
    });
    expect(toast.success).toHaveBeenCalledWith(
      '¡Usuario registrado exitosamente! Ya podés iniciar sesión con estas credenciales'
    );
  });
});
