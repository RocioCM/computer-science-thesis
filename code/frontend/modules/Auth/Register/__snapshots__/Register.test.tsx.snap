// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page Matches snapshot 1`] = `
<div>
  <main
    class="w-full h-screen bg-p1-25 flex flex-row items-center justify-around gap-xl p-3xl"
  >
    <img
      alt="Iniciar sesión"
      class="absolute h-full w-full object-cover opacity-90 z-0"
      data-nimg="1"
      decoding="async"
      height="800"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=1200&q=75 1x, /_next/image?url=%2Fimg.jpg&w=3840&q=75 2x"
      style="color: transparent;"
      width="1200"
    />
    <form
      class="relative w-full min-w-[300px] max-w-[500px] h-max max-h-full overflow-auto rounded-rs border border-n2 bg-n0 shadow-e3 p-2xl flex flex-col gap-xl"
    >
      <h2
        class="font-medium"
      >
        Registro
      </h2>
      <div
        class="flex flex-col gap-s"
      >
        <div
          class="w-full"
        >
          <span
            class="block relative z-10 w-max transition-all text-p mb-xs top-0 left-0 text-n4 font-medium"
          >
            Correo electrónico
          </span>
          <div
            class="relative flex flex-row flex-nowrap justify-between items-center w-full"
          >
            <div
              class="absolute left-0 h-full w-max flex flex-row items-center gap-xs"
            />
            <input
              class="input block w-full relative border bg-transparent outline-none transition-shadow p rounded-rs border-n3 px-m py-xs placeholder:text-n2 h-10 hover:border-n2 inputFocus focus:border-p1 focus:border-2 disabled:text-n3 disabled:bg-n1 disabled:border-n2 disabled:placeholder:text-transparent"
              data-testid="input"
              form="[object Object]"
              inputlabel="Correo electrónico"
              name="email"
              options=""
              placeholder=""
              required=""
              type="text"
              value=""
            />
            <div
              class="absolute right-0 h-full w-max flex flex-row items-center gap-xs"
            />
          </div>
        </div>
        <div
          class="w-full"
        >
          <span
            class="block relative z-10 w-max transition-all text-p mb-xs top-0 left-0 text-n4 font-medium"
          >
            Contraseña
          </span>
          <div
            class="relative flex flex-row flex-nowrap justify-between items-center w-full"
          >
            <div
              class="absolute left-0 h-full w-max flex flex-row items-center gap-xs"
            />
            <input
              class="input block w-full relative border bg-transparent outline-none transition-shadow p rounded-rs border-n3 px-m py-xs placeholder:text-n2 h-10 hover:border-n2 inputFocus focus:border-p1 focus:border-2 disabled:text-n3 disabled:bg-n1 disabled:border-n2 disabled:placeholder:text-transparent pr-10"
              data-testid="input"
              form="[object Object]"
              inputlabel="Contraseña"
              name="password"
              options=""
              placeholder=""
              required=""
              type="password"
              value=""
            />
            <div
              class="absolute right-0 h-full w-max flex flex-row items-center gap-xs"
            >
              <img
                alt="Show password"
                class="w-l h-l cursor-pointer mr-s"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                role="button"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </div>
          </div>
        </div>
        <div
          class="w-full"
        >
          <span
            class="block relative z-10 w-max transition-all text-p mb-xs cursor-pointer top-0 left-0 text-n4 font-medium"
          >
            Rol
          </span>
          <button
            class="relative flex flex-row flex-nowrap justify-between items-center w-full border bg-transparent outline-none transition-shadow gap-xs p text-left rounded-rs border-n3 px-m py-m placeholder:text-n2 h-10 hover:border-n2 dropdonwFocus focus:border-p1 focus:border-2 disabled:text-n3 disabled:bg-n1 disabled:border-n2"
            data-testid="dropdown"
            name="roleId"
          >
            <span
              class="flex-1 min-h-4"
            >
              Selecciona un rol
            </span>
            <i
              class="inline-block w-2.5 h-2.5 rotate-0 "
              data-testid="chevron-icon"
            >
              <svg
                class="w-full h-full block transition-transform duration-300 ease-out"
                fill="none"
                viewBox="0 0 15 9"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14.7 1.89992L13.5 0.919922L7.50001 6.09992L1.50001 0.919921L0.300013 1.89992L7.50001 8.19992L14.7 1.89992Z"
                  fill="var(--colors-neutrals-n10)"
                />
              </svg>
            </i>
          </button>
        </div>
      </div>
      <p
        class="text-center"
      >
        ¿Ya tienes cuenta?
         
        <a
          class="font-semibold text-p1"
          href="/auth/login"
        >
          Ingresa aquí
        </a>
      </p>
      <button
        class="relative flex items-center justify-center gap-s rounded-rs font-semibold h-10 disabled:cursor-default disabled:bg-opacity-40  focusButton bg-p1 text-n0 py-s px-m w-full"
        data-testid="button"
        disabled=""
      >
        <span>
          Crear usuario
        </span>
      </button>
    </form>
  </main>
</div>
`;
